### PR TITLE
ut: init at 2.0.0

### PR DESCRIPTION
**Pull request by MatthewCroughan**
## Description of changes

Initialise https://github.com/boost-ext/ut at 2.0.0 in Nixpkgs

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ut/ut/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  version = "2.0.0";
+
+  cmakeFlags = [
+    "-DBOOST_UT_ALLOW_CPM_USE=OFF"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "boost-ext";
+    repo = "ut";
+    rev = "v${version}";
```
Comment (ambroisie) [inline pkgs/by-name/ut/ut/package.nix:18]: Use the [`finalAttrs` idiom](https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes).

**Review by ambroisie — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ut/ut/package.nix (hunk 1 above)
Comment (MatthewCroughan) [reply pkgs/by-name/ut/ut/package.nix]: Done.

**Review by ambroisie — APPROVED** [review]

File: pkgs/by-name/ut/ut/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-nMRhVtbnORMnZxIdrv6pvfj8/fv/buh4Y7mz0cr4p+E=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  meta = with lib; {
+    description = "UT: C++20 μ(micro)/Unit Testing Framework";
```
Comment (MikaelFangel) [inline pkgs/by-name/ut/ut/package.nix:28]:
```suggestion
    description = "C++20 μ(micro)/Unit Testing Framework";
```

File: pkgs/by-name/ut/ut/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    homepage = "https://github.com/boost-ext/ut";
+    license = licenses.boost;
+    maintainers = with maintainers; [ matthewcroughan ];
+    platforms = platforms.all;
```
Comment (MikaelFangel) [inline pkgs/by-name/ut/ut/package.nix:32]:
```suggestion
    platforms = platforms.all;
    mainProgram = "ut";
```

File: pkgs/by-name/ut/ut/package.nix (hunk 2 above)
Comment (MatthewCroughan) [reply pkgs/by-name/ut/ut/package.nix:28]: Done

File: pkgs/by-name/ut/ut/package.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    mainProgram = "ut";
```
Comment (pbsds) [inline pkgs/by-name/ut/ut/package.nix:33]:
```suggestion
    mainProgram = "ut";
```

This binary does not exist?

Comment (ambroisie) [reply pkgs/by-name/ut/ut/package.nix:33]: Yes, as pointed out in [my earlier comment](https://github.com/NixOS/nixpkgs/pull/265978#discussion_r1385580737)

Comment (MikaelFangel) [reply pkgs/by-name/ut/ut/package.nix:33]: Oh, I actually never thought that this got committed, based on the comment. But it should only have been added if there was a binary matching 😅

Comment (MatthewCroughan) [reply pkgs/by-name/ut/ut/package.nix]: Fixed.

**Pull request merged**